src/Microsoft.AspNet.Hosting/Program.cs

```diff
@@ -15,21 +17,24 @@ public class Program
         private const string HostingJsonFile = "Microsoft.AspNet.Hosting.json";
         private const string ConfigFileKey = "config";
 
-        private readonly IServiceProvider _serviceProvider;
-
-        public Program(IServiceProvider serviceProvider)
-        {
-            _serviceProvider = serviceProvider;
-        }
-
-        public void Main(string[] args)
+        public static void Main(string[] args)
         {
             // Allow the location of the json file to be specified via a --config command line arg
             var tempBuilder = new ConfigurationBuilder().AddCommandLine(args);
             var tempConfig = tempBuilder.Build();
             var configFilePath = tempConfig[ConfigFileKey] ?? HostingJsonFile;
 
-            var appBasePath = _serviceProvider.GetRequiredService<IApplicationEnvironment>().ApplicationBasePath;
+            var services = new ServiceCollection();
+            services.TryAdd(ServiceDescriptor.Instance(PlatformServices.Default.Application));
+            services.TryAdd(ServiceDescriptor.Instance(PlatformServices.Default.Runtime));
+            services.TryAdd(ServiceDescriptor.Instance(PlatformServices.Default.AssemblyLoadContextAccessor));
+            services.TryAdd(ServiceDescriptor.Instance(PlatformServices.Default.AssemblyLoaderContainer));
+            services.TryAdd(ServiceDescriptor.Instance(PlatformServices.Default.LibraryManager));
+            services.TryAdd(ServiceDescriptor.Instance(CompilationServices.Default.LibraryExporter));
+            var serviceProvider = services.BuildServiceProvider();
```

Comment:
@davidfowl Since this is all static now can we just move it into a WebHostBuilder constructor? That should also fix TestHost.